### PR TITLE
fauxton-visual-guide

### PR DESCRIPTION
**Pull request by michellephung**
Trying to add this:
http://michellephung.github.io/Fauxton-Visual-Guide/
to the site. Not sure if this is how I should do it, since make didn't make for me.

### CODE REVIEW

**Comment by kxepal** [comment]
You should use official CouchDB logo for speaking couch. Imagine that those person speaks.

**Comment by kxepal** [comment]
TOC link is too pale and it's easy to miss it.

**Comment by kxepal** [comment]
`(download available here!)` could reference directly to download section.

**Comment by kxepal** [comment]
While in overview you provides a link to CouchDB 1.6.1 documentation, all API reference goes to latest version which is for 2.0.

**Comment by kxepal** [comment]
Grey text on grey background hard to read. May be make it a bit bolder and blacker?

**Comment by kxepal** [comment]
```
Active Tasks Displays a list of the running background tasks on the server. Background tasks include view index building, compaction and replication. This page is an interface to the Active Tasks API call.

Config An interface into the configuration of your CouchDB installation. The interface allows you to edit the different configurable parameters. For more details on configuration, see Configuring CouchDB.
```

There are no delimiters between Active Tasks and Config and the rest of text. May be rephrase text to better integrate internal links into it or add some explicit delimiter like `-` or `:`?

UPD:

Well, here is indeed you can embed internal link with easy:

```
Verify The Verify Installation allows you to check whether all of the components of your CouchDB installation are correctly installed.
```

**Comment by kxepal** [comment]
- this image is very hard to read.

**Comment by kxepal** [comment]
On replication section figures 1 and 2 on image are not explained.

**Comment by kxepal** [comment]
It's quite strange to say "Hi" at the end of the page.

**Comment by kxepal** [comment]
In everything else - awesome work!

**Comment by michellephung** [comment]
@kxepal I pushed changes per your feedback onto my github.io page, and to this branch. I think I got all the links you were talking about (I hope) and they now point to latest version instead of 1.6.1. 

Let me know if you like these new colors better :) I darkened the TOC, but its still a little light in color, because I don't want it to be too obtrusive while reading the rest of the page. 

The header bar diagram: :/ I don't like this graph too much either. I went through a handful of versions for this diagram, and maybe I have been looking at it too long. I will have to think about it some more. If you have any suggestions let me know please.

I didn't want to say 'good-bye' at the end. So I just said Hi! instead!
Suggestions are also welcome here, but i do rather like the surprise of the unexpected strangeness. I was restraining how strange I wanted to be throughout the page, and I held back, I think fairly well, until the end.

Hi!

File: fauxton-visual-guide/main.css [hunk 1]
```diff
@@ -0,0 +1 @@
...ck;margin-left:auto;margin-right:auto}.page img.couch{height:auto;position:relative;top:50%;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%)}.leftside{width:50%;float:left}.rightside{color:green;width:50%;float:right;height:100%}.linen{background:url(assets/linen.jpg) #eee 50% 0%}.textbox{color:gray;border:1px solid #d8d8d8;min-height:600px;width:87%;margin:auto;margin-top:70px;margin-bottom:50px;background-color:rgba(242,242,242,0.85);overflow:hidden}.textbox h1{font-size:30px;text-align:center;margin:20px 10px;color:#B5B5B5}.textbox p{padding:5px 30px 15px 30px;line-height:1.5}.textbox p.br{padding:0 30px;line-height:1}.textbox p.indent{margin-left:15px}.textbox p.question{font-weight:600;padding-bottom:0}.textbox p.answer{padding-top:0}#introBox{text-align:center;min-height:320px;display:block;position:relative;top:50%;-webkit-transform:translateY(50%);-ms-transform:translateY(50%);transform:translateY(50%)}#introBox h1{margin-top:100px}#introBox h1{font-size:60px;color:#C0C0C0}#introBox h2{font-size:16px;padding:0}#page-overview img{margin-top:300px;height:auto}#page-header-bar img{padding:30px;height:800px}#page-answers img{height:auto}
```
Comment (robertkowalski) [inline fauxton-visual-guide/main.css]: this looks like compiled output, where is the source for it?

File: fauxton-visual-guide/js/main.js [hunk 2]
```diff
@@ -0,0 +1 @@
+$(function(){var t,n;n=$("#toc");t=$("#toggle-toc-button");return t.click(function(){n.toggleClass("hidden");t.toggleClass("x");return t.text(function(){if(n.hasClass("hidden")){return"TOC"}else{return"X"}})})});
```
Comment (robertkowalski) [inline fauxton-visual-guide/js/main.js:1]: this looks like compiled output, where is the source for it?

File: fauxton-visual-guide/js/jquery.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
...his.now=this.cur(),this.end=d,this.unit=f||(m.cssNumber[c]?"":"px")},cur:function(){var a=Za.propHooks[this.prop];return a&&a.get?a.get(this):Za.propHooks._default.get(this)},run:function(a){var b,c=Za.propHooks[this.prop];return this.options.duration?this.pos=b=m.easing[this.easing](a,this.options.duration*a,0,1,this.options.duration):this.pos=b=a,this.now=(this.end-this.start)*b+this.start,this.options.step&&this.options.step.call(this.elem,this.now,this),c&&c.set?c.set(this):Za.propHooks._default.set(this),this}},Za.prototype.init.prototype=Za.prototype,Za.propHooks={_default:{get:function(a){var b;return null==a.elem[a.prop]||a.elem.style&&null!=a.elem.style[a.prop]?(b=m.css(a.elem,a.prop,""),b&&"auto"!==b?b:0):a.elem[a.prop]},set:function(a){m.fx.step[a.prop]?m.fx.step[a.prop](a):a.elem.style&&(null!=a.elem.style[m.cssProps[a.prop]]||m.cssHooks[a.prop])?m.style(a.elem,a.prop,a.now+a.unit):a.elem[a.prop]=a.now}}},Za.propHooks.scrollTop=Za.propHooks.scrollLeft={set:function(a){a.elem.nodeType&&a.elem.parentNode&&(a.elem[a.prop]=a.now)}},m.easing={linear:function(a){return a},swing:function(a){return.5-Math.cos(a*Math.PI)/2}},m.fx=Za.prototype.init,m.fx.step={};var $a,_a,ab=/^(?:
```
Comment (robertkowalski) [inline fauxton-visual-guide/js/jquery.js]: We already have jquery included on the page i think

**Comment by robertkowalski** [comment]
@michellephung can we change our makefile to something like (caution, untested)

``` Makefile
all: style/master.css fauxton-visual-guide

style/master.css: style/master.less
    lessc $< > $@

fauxton-visual-guide:
    harp compile src-fauxton-visual-guide fauxton-visual-guide

clean:
    rm style/master.css
    rm -rf fauxton-visual-guide
```

and check in the sources of the folder `src-fauxton-visual-guide` which then compiles to fauxton-visual-guide when we run `make`

**Comment by michellephung** [comment]
i can will work more on the diagram and the make file, for getting it in a better place to compile/work with for this pr on friday or monday.

could you guys look over the cors section. I think I wrote it wrong. I'm sorry. :worried:

File: fauxton-visual-guide/index.html [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+database name (either source or target) using these tabs.</p>
+</div></div></div><div id="page-config" class="pagecontainer"><div class="leftside page"><img src="assets/config.png"></div><div class="rightside linen page"><div class="textbox"><h1 id="configuration">Configuration</h1>
+<p>An interface into the configuration of your CouchDB installation. The 
+interface allows you to edit the different configurable parameters. For 
+more details on configuration, see the 
+<a href="http://docs.couchdb.org/en/latest/config/index.html#config">Configuring CouchDB</a> 
+section. </p>
+</div></div></div><div id="page-cors" class="pagecontainer"><div class="leftside page"><img src="assets/cors.png"></div><div class="rightside linen page"><div class="textbox"><h1 id="cors">CORS</h1>
+<p>CORS stands for Cross Origin Resource Sharing. It is setting that 
+pertains only to browsers.</p>
```
Comment (robertkowalski) [inline fauxton-visual-guide/index.html:137]: isn't it "It is a setting.." ?

File: fauxton-visual-guide/index.html [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+<p>By enabling CORS, you allow your browser to access data from other 
+people, recieving from and sending the data to <i>their</i> computers. 
+For example, you can ping someone else&#39;s CouchDB instance through an 
+AJAX request, and it will return information. Or you can POST data from 
+your browser or site, and enabling CORS allows you to send that 
+information.</p>
```
Comment (robertkowalski) [inline fauxton-visual-guide/index.html]: maybe this

By enabling CORS, you allow requests that the browser sends from another domain to access/modify the data of your CouchDB. For example, if your CouchDB installation is running on the domain alice.example you can ping someone else's CouchDB instance through an AJAX request from a different domain, e.g. bob.example - or even localhost! 

But be careful - allowing any domain to access your data can be a security risk! If you only want certain websites (origins) to be able to retrieve and change data, you can set &#39;Restrict to specific domains&#39;, and then list which origins you want to allow access for.

If you disable CORS the JavaScript that makes the AJAX request needs to run on the same domain. With CORS disabled frontend code that runs on alice.example must be also hosted on alice.example.

File: fauxton-visual-guide/index.html (hunk 4 above)
Comment (michellephung) [inline fauxton-visual-guide/index.html:137]: yes

Comment (michellephung) [inline fauxton-visual-guide/index.html:137]: good catch

File: fauxton-visual-guide/index.html [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+people you can recieve data from. If you only want certain websites ( 
+origins) to be able to retrieve data from, you can set 
+&#39;Restrict to specific domains&#39;, and then list which origins you want to 
+allow access for.</p>
+<p>By setting, &#39;All domains&#39;, you allow receiving and sending information 
+to anyone.</p>
+<p>This is a browser specific layer of protection, a second layer of 
+protection, if you will. The first layer of protection comes from the 
+permissions you set on each database, and whether they have _reader, 
+_writer, _admin, etc. access.</p>
```
Comment (robertkowalski) [inline fauxton-visual-guide/index.html]: I think CouchDB does not know _reader and _writer out of the box, just admins and members.

```
The first layer of protection comes from the permissions you set on each database using roles.
```

and then link http://docs.couchdb.org/en/latest/api/database/security.html

**Comment by benkeen** [comment]
Slick! Very nice job, @michellephung.
- "Troubleshooting" seems the wrong title for that page. Maybe "Helpful Links" and drop the second line.
- Could we add a "How to install / setup" link to that page. Seems significant enough to place first.

File: fauxton-visual-guide/js/main.js (hunk 2 above)
Comment (michellephung) [inline fauxton-visual-guide/js/main.js:1]: this is how i write code. what?

Comment (kxepal) [inline fauxton-visual-guide/js/main.js:1]: Turing test passed (:

Comment (michellephung) [inline fauxton-visual-guide/js/main.js:1]: :D

**Comment by chewbranca** [comment]
This is awesome! Great work @michellephung!

My only suggestion is in the `All Documents - The Header Bar` section to is to vertically align the numbered steps with the appropriate images, otherwise the step descriptions can be off screen when you look at the bottom images.

**Comment by michellephung** [comment]
I am still working on this! I needs more time.

**Comment by drsm79** [comment]
Super nice - good job :)

**Pull request merged**